app/src/ui/dispatcher/dispatcher.ts

```diff
@@ -970,6 +970,38 @@ export class Dispatcher {
     mergeStatus: MergeTreeResult | null,
     isSquash: boolean = false
   ): Promise<void> {
+    const {
+      multiCommitOperationState,
+      branchesState: { tip },
+    } = this.repositoryStateManager.get(repository)
+
+    if (
+      multiCommitOperationState === null ||
+      multiCommitOperationState.operationDetail.kind !==
+        MultiCommitOperationKind.Merge
+    ) {
```

Comment:
Feels weird having this kind of check here knowing `mergeBranch` is also used in other merge entry points to do the actual merge (whereas the multi commit operation was initialized somewhere else). I think I'd rather have a new method in the dispatcher for this that `updateBranch` in app.tsx would call, but this is ok.

In any case, I think it'd be good adding a comment right above this `if` to explain why sometimes there is an ongoing Merge operation, and sometimes there isn't 🤔

Reply:
Yeah. I like that better.. this actually broke in 2 places. updateBranch and in the compare mode.